app/upgrades/mainnet/v4/upgrades.go

```diff
@@ -0,0 +1,91 @@
+package v4
+
+import (
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	slashingkeeper "github.com/cosmos/cosmos-sdk/x/slashing/keeper"
+	upgradetypes "github.com/cosmos/cosmos-sdk/x/upgrade/types"
+	ibcclientkeeper "github.com/cosmos/ibc-go/v3/modules/core/02-client/keeper"
+	ibcclienttypes "github.com/cosmos/ibc-go/v3/modules/core/02-client/types"
+
+	feemarketkeeper "github.com/tharsis/ethermint/x/feemarket/keeper"
+	feemarkettypes "github.com/tharsis/ethermint/x/feemarket/types"
+
+	claimskeeper "github.com/tharsis/evmos/v3/x/claims/keeper"
+)
+
+// CreateUpgradeHandler creates an SDK upgrade handler for v2
+func CreateUpgradeHandler(
+	mm *module.Manager,
+	configurator module.Configurator,
+	clientKeeper ibcclientkeeper.Keeper,
+	slashingKeeper slashingkeeper.Keeper,
+	feemarketKeeper feemarketkeeper.Keeper,
+	claimsKeeper *claimskeeper.Keeper,
+) upgradetypes.UpgradeHandler {
+	return func(ctx sdk.Context, _ upgradetypes.Plan, vm module.VersionMap) (module.VersionMap, error) {
+		// Refs:
+		// - https://docs.cosmos.network/master/building-modules/upgrade.html#registering-migrations
+		// - https://docs.cosmos.network/master/migrations/chain-upgrade-guide-044.html#chain-upgrade
+
+		updateSlashingParams(ctx, slashingKeeper)
+		if err := updateIBCClients(ctx, clientKeeper); err != nil {
+			return vm, err
+		}
+
+		updateFeeMarketParams(ctx, feemarketKeeper)
+		updateClaimsParams(ctx, claimsKeeper)
+
+		// Leave modules are as-is to avoid running InitGenesis.
+		return mm.RunMigrations(ctx, configurator, vm)
+	}
+}
+
+// updateSlashingParams changes the slashing window from 30k to 10k
+func updateSlashingParams(ctx sdk.Context, k slashingkeeper.Keeper) {
+	params := k.GetParams(ctx)
```

Comment:
didn't this cause problems last time, we tried to upgrade? I remember we had to compose the params manually instead of getting them from the store.

Reply:
Good question. This was in #363 

But that was a problem during a store migration

I think it is safe to do in this upgrade handler that doesn't involve a store migration, as shown by: 
https://github.com/osmosis-labs/osmosis/blob/main/app/upgrades/v7/upgrades.go#L37

Reply:
> didn't this cause problems last time, we tried to upgrade? I remember we had to compose the params manually instead of getting them from the store.

correct, the main difference here is that we are not introducing a new parameter to the modules, so this time the function won't panic. I'll add tests for this upgrade logic